compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -2295,6 +2307,56 @@ private BLangUnionTypeNode addUnionType(BLangType lhsTypeNode, BLangType rhsType
         return unionTypeNode;
     }
 
+    private boolean recordTypeDescritorIsAnonymous(RecordTypeDescriptorNode recordTypeDescriptorNode) {
+        Node parent = recordTypeDescriptorNode.parent();
+        switch (parent.kind()) {
+            case UNION_TYPE_DESC:
+            case ARRAY_TYPE_DESC:
+            case OPTIONAL_TYPE_DESC:
+            case TYPE_TEST_EXPRESSION:
+                return true;
+            default:
+                return false;
+        }
+    }
+
+    private boolean ifInLocalContext(Node parent) {
+        while (parent != null) {
+            if (parent instanceof StatementNode) {
```

Comment:
is it possible to replace this with a `.kind()` check?

Reply:
fixed with 0f3b802. Didnt use the `kind()` check since there were a lot of types to check. used a flag instead.